components/bookshelf/discover/SearchBox.tsx

```diff
@@ -16,19 +23,26 @@ interface SearchBoxProps {
 const SearchBox = ({ setIsSearching }: SearchBoxProps) => {
   const [filteredData, setFilteredData] = useState()
   const [searchInput, setSearchInput] = useState()
-  const context = useContext(BooksContext)
-  if (!context) {
-    return null
-  }
-  const { books } = context
+  const [books, setBooks] = useState<BookType[]>([])
+  const { user } = useContext(UserContext)
+
+  useEffect(() => {
+    const fetchBooks = async () => {
+      const bookList = await getBookList(user)
+      setBooks(bookList)
+    }
+
+    fetchBooks().catch(console.error)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
 
   const handleChange = (e) => {
     setSearchInput(e.target.value)
   }
 
   const handleSubmit = (e) => {
     e.preventDefault()
-    const filteredBooks = books.filter((book) => {
+    const filteredBooks = books.filter((book: BookType) => {
```

Comment:
Isn't `: BookType` already inferred by typescript as `BookType`? Since `books` is a state with type `BookType[]`?